src/pudl_archiver/depositors/fsspec.py

```diff
@@ -0,0 +1,272 @@
+"""Depositor implementation using any fsspec compatible file-system as storage backend."""
+
+import logging
+import traceback
+from typing import BinaryIO
+
+import aiohttp
+from pydantic import BaseModel, ConfigDict
+from upath import UPath
+
+from ..frictionless import MEDIA_TYPES, DataPackage, Resource, ResourceInfo
+from ..utils import RunSettings, compute_md5
+from .depositor import (
+    DepositionAction,
+    DepositionChange,
+    DepositorAPIClient,
+    DraftDeposition,
+    PublishedDeposition,
+    register_depositor,
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+
+def _resource_from_upath(path: UPath, parts: dict[str, str]) -> Resource:
+    """Create a resource from a single file with partitions.
+
+    Args:
+        file: Deposition file metadata returned by Zenodo api.
```

Comment:
This seems like leftover cruft from Zenodo, as it's not a required input. Conversely looks like `path` is missing here.

Reply:
FIxed!